File: 003/euler3.js
function factorize(n) {
  var factors = [];
  var m = 1, max = Math.ceil(Math.sqrt(n));
  while (++m < max) if (n % m == 0) factors.push(m);
  return factors;
}


function f(n) {
  return factorize(n).filter(function(factor) {
    return factorize(factor).length == 0;
  });
}


if (process.argv[1] == __filename) {
  require('assert').deepEqual(f(13195), [5, 7, 13, 29]);
  console.log(f(600851475143).slice(-1)[0]);
}

File: 003/euler3.py
import math


isqrt = lambda n: int(math.ceil(math.sqrt(n)))

factorize = lambda n: [m for m in range(2, isqrt(n)) if n % m == 0]

f = lambda n: [m for m in factorize(n) if not factorize(m)]


if __name__ == '__main__':
    assert f(13195) == [5, 7, 13, 29]
    print(f(600851475143)[-1])
